ksqldb-engine/src/test/java/io/confluent/ksql/function/udaf/offset/LatestByOffsetUdafTest.java

```diff
@@ -96,57 +209,146 @@ public void shouldMergeWithOverflow() {
     assertThat(merged2, is(agg2));
   }
 
+  @Test
+  public void shouldMergeWithOverflowNIntegers() {
+    // Given:
+    final Udaf<Integer, List<Struct>, List<Integer>> udaf = LatestByOffset.latestIntegers(2);
+
+    LatestByOffset.sequence.set(Long.MAX_VALUE - 1);
+
+    Struct struct1 = LatestByOffset.createStruct(STRUCT_INTEGER, 123);
+    Struct struct2 = LatestByOffset.createStruct(STRUCT_INTEGER, 321);
+    Struct struct3 = LatestByOffset.createStruct(STRUCT_INTEGER, 543);
+    Struct struct4 = LatestByOffset.createStruct(STRUCT_INTEGER, 654);
+    
+    List<Struct> agg1 = Lists.newArrayList(struct1, struct2);
+    List<Struct> agg2 = Lists.newArrayList(struct3, struct4);
+
+    // When:
+    final List<Struct> merged1 = udaf.merge(agg1, agg2);
+    final List<Struct> merged2 = udaf.merge(agg2, agg1);
+
+    // Then:
+    assertThat(agg1.get(0).getInt64(SEQ_FIELD), is(Long.MAX_VALUE - 1));
+    assertThat(agg2.get(0).getInt64(SEQ_FIELD), is(Long.MIN_VALUE));
+    assertThat(merged1, contains(struct3, struct4));
+    assertThat(merged2, contains(struct3, struct4));
+  }
 
   @Test
   public void shouldComputeLatestLong() {
     // Given:
     final Udaf<Long, Struct, Long> udaf = LatestByOffset.latestLong();
 
     // When:
-    Struct res = udaf
-        .aggregate(123L, LatestByOffset.createStruct(STRUCT_LONG, 321L));
+    Struct res = udaf.aggregate(123L, LatestByOffset.createStruct(STRUCT_LONG, 321L));
 
     // Then:
     assertThat(res.getInt64(VAL_FIELD), is(123L));
   }
+  
+  @Test
+  public void shouldComputeLatestNLongs() {
+    // Given:
+    final Udaf<Long, List<Struct>, List<Long>> udaf = LatestByOffset.latestLong(3);
+
+    // When:
+    List<Struct> res = udaf
+        .aggregate(123L, Lists.newArrayList(LatestByOffset.createStruct(STRUCT_LONG, 321L)));
+    
+    List<Struct> res2 = udaf
+        .aggregate(543L, res);
+    
+    List<Struct> res3 = udaf
+        .aggregate(654L, res);
```

Comment:
Fix spot bugs issue:

Either asset the contents of `res` and `res2` or remove them as variables.